### PR TITLE
Create a multi-architecture Docker image for the runtime

### PR DESCRIPTION
**Pull request by jonpspri**
Use gradle-docker-plugin and (my) gradle-fatmanifest-plugin to create a multi-architecture manifest for the nodejs runtime.  The process relies on having PPC64LE and S390X docker environments available.  The TLS certificates for these environments are encrypted in the repo.  The GPG key for them has been encrypted as well, accessible by my fork and by ‘apache/incubator-openwhisk-runtime-nodejs’.  Other forks can be given access — it is necessary to add encryption’s of the GPG key and the JSON specifying the repository locations in the .travis.yml.  They will also need to provide their own Docker registry (or we need to do some additional work to support using an ephemeral registry — the problem is that we then need to add a capability to the gradle-fatmanifest-plugin to copy blobs between registries, not a pretty proposition. 

I’m going to try to add some documentation of the process to the files and to a docs folder, but I wanted to get review started now, as I suspect there’ll be many questions and those will in turn inform the documentation.

This is a key milestone in being able to run OpenWhisk in a multi-architecture environment.  Being able to take advantage of it will eventually depend upon [OpenWhisk Issue 3412](https://github.com/apache/incubator-openwhisk/pull/3412) and other following changes to do away with hard-coding.  (See [OpenWhisk PR 3407](https://github.com/apache/incubator-openwhisk/pull/3407) for an outline of upcoming changes.)

There is a severe backlog of cleanup coming, too.  Issue to follow to document needed cleanup.

@csantanapr

### CODE REVIEW

**Comment by jonpspri** [comment]
Hmm... as expected, the encrypted keys are presenting some difficulty, in part because of quoting issues.  I’ll look at it more closely in the morning when I have slept.

**Comment by jonpspri** [comment]
Re-based and removed superfluous changes to testing processes.

**Comment by jonpspri** [comment]
@csantanapr @rabbah Could you give this a review?  Here's the follow-on items list:
- Delete `docker.gradle` and various unused `docker.ignore` files
- Refactor the tests to use the common modules we're creating in https://github.com/apache/incubator-openwhisk/pull/3467
- Remove the need to build and deploy `invoker-openwhisk` in order to run the tests

File: .gitignore [hunk 1]
```diff
@@ -63,11 +63,8 @@ ansible/roles/nginx/files/*cert.pem
 
 # .zip files must be explicited whitelisted
 *.zip
-!tests/dat/actions/blackbox.zip
```
Comment (rabbah) [inline .gitignore]: because these dont exist in this repo

File: build.gradle [hunk 2]
```diff
@@ -1,9 +1,306 @@
...
+        classpath 'com.bmuschko:gradle-docker-plugin:3.2.2'
+        classpath 'com.xanophis:gradle-fatmanifest-plugin:0.0.5'
+    }
+}
+
+import com.bmuschko.gradle.docker.DockerRegistryCredentials
+import com.bmuschko.gradle.docker.tasks.DockerInfo
+import com.bmuschko.gradle.docker.tasks.image.*
+
+//  There are some serious naming mismatchees to be dealt with in v0.0.6 of gradle-fatmanifest-plugin :(
```
Comment (rabbah) [inline build.gradle]: mismatchees typo

File: .gitignore (hunk 1 above)
Comment (jonpspri) [reply .gitignore]: Is there an emoji for “Hey I just left it there because you guys always knock me for changing too many things at once?”. 😛

**Pull request closed without merging**